include/mrdocs/GeneratorWithTagfile.hpp

```diff
@@ -0,0 +1,89 @@
+//
+// This is a derivative work. originally part of the LLVM Project.
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// Copyright (c) 2024 Fernando Pelliccioni (fpelliccioni@gmail.com)
+//
+// Official repository: https://github.com/cppalliance/mrdocs
+//
+
+#ifndef MRDOCS_API_GENERATOR_WITH_TAGFILE_HPP
+#define MRDOCS_API_GENERATOR_WITH_TAGFILE_HPP
+
+#include <mrdocs/Config.hpp>
+#include <mrdocs/Corpus.hpp>
+#include <mrdocs/Generator.hpp>
+#include <mrdocs/Platform.hpp>
+#include <mrdocs/Support/Error.hpp>
+
+#include <lib/Lib/TagfileWriter.hpp>
+
+#include <ostream>
+#include <string_view>
+
+namespace clang {
+namespace mrdocs {
+
+/** Base class for generators that also create tagfiles.
+    This class provides an additional API to handle
+    the generation of tagfiles alongside the main documentation output.
+*/
+class GeneratorWithTagfile
```

Comment:
Better than before, which was breaking the API. Recently, we merged all the code from `adoc` and `html` because the only difference between them was the file extensions. Now that we have simplified the generators into HandlebarsGenerator, this can be even simpler because these latest changes made this class redundant. 

```cpp
build(std::string_view outputPath,Corpus const& corpus)
```

is the function we use from `GenerateAction`. And it's already part of the contract that it might generate multiple files (because multi-page). 

`buildOne` is what we use for tests, and it can't generate multiple outputs.

So now the `generateTagfile(...)` logic can go in an anonymous namespace in `HandlebarsGenerator.cpp` (so no need to have another Generator class and not breaking the contract of `build` for the handlebars generator), and the logic of `HandlebarsGenerator::build` only needs to become:

```
     if (!corpus.config->multipage)
    {
        Error e = Generator::build(outputPath, corpus);
        if (!e.failed() && !corpus.config->tagfilePath.empty())
        {
            return generateTagfile(...);
        }
    }
// ...
    auto errors = ex->wait();
    if (!errors.empty())
    {
        return Error(errors);
    }
    if (!corpus.config->tagfilePath.empty())
    {
          return generateTagfile(...);
    }
    return Error::success();
```

This anon function `generateTagfile` can take whatever you need, which is probably just the dom corpus and the file extension.